clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -1793,6 +1793,37 @@ void ItaniumCXXABI::EmitDestructorCall(CodeGenFunction &CGF,
                             ThisTy, VTT, VTTTy, nullptr);
 }
 
+// Check if any non-inline method has the specified attribute.
+template <typename T>
+static bool CXXRecordNonInlineHasAttr(const CXXRecordDecl *RD) {
+  for (const auto *D : RD->noload_decls()) {
+    if (const auto *FD = dyn_cast<FunctionDecl>(D)) {
+      if (FD->isInlineSpecified() || FD->doesThisDeclarationHaveABody() ||
```

Comment:
Is there some reason you're not using `FD->isInlined()` here?

Reply:
This was brought over from the original code and I assumed that this is what is meant by "inline" in this context.

Reply:
"isInlineSpecified()" specifically refers to the keyword.  `isInlined()` is what the C++ standard calls an "inlined function", and it includes some other stuff: constexpr functions, and functions where the body is written inside the class definition.

Reply:
Yes, I understand that they are not equivalent but the original code has used `isInlineSpecified()` for quite some time (has been downstream at Sony since 2015). Are you OK with this patch going in as is with a follow up to look into using `isInlined()`?

Reply:
@wjristow, I believe that you wrote the original code. Do you remember why `isInlineSpecified()` was used rather than `isInlined()`?

Reply:
It's probably fine to merge as-is, sure; inline functions marked with import/export attributes should be pretty rare in any case.

Reply:
> Do you remember why `isInlineSpecified()` was used rather than `isInlined()`?

No, I don't recall.  I've reviewed the relevant internal ticket, and I don't see any reason in the notes there for it, either.  Most likely it was just a mistake I made back in 2015.  So I think it makes sense to change it to use `isInlined()`.

Reply:
OK, I have changed `isInlineSpecified()` to `isInlined()`. I re-ran some of our downstream testing and confirmed there was no negative impact.